segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,170 @@
+                 segmentation_id: int,
+                 roi_id: int):
+        self.image_shape = image_shape
+        self.segmentation_id = segmentation_id
+        self.roi_id = roi_id
+        self._sparse_coo = coo_matrix((coo_data, (coo_rows, coo_cols)),
+                                      shape=image_shape)
+
+    @classmethod
+    def roi_from_query(cls, segmentation_id:int,
+                       roi_id: int) -> "ROI":
+        """
+        Queries and builds ROI object by querying LIMS table for
+        produced labeling ROIs.
+        Args:
+            segmentation_id: Id of the segmentation run
+            roi_id: Unique Id of the ROI to be loaded
+
+        Returns: ROI object for the given segmentation_id and roi_id
+        """
+        label_vars = query_utils.get_labeling_env_vars()
+
+        shape = query_utils.query(
+            f"SELECT * FROM public.segmentation_runs WHERE id={segmentation_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)[0]['video_shape']
+
+        roi = query_utils.query(
+            f"SELECT * FROM public.rois WHERE "
+            f"segmentation_run_id={segmentation_id} AND id={roi_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)
+        return ROI(coo_rows=roi['coo_rows'][0],
+                   coo_cols=roi['coo_cols'][0],
+                   coo_data=roi['coo_data'][0],
+                   image_shape=shape,
+                   segmentation_id=segmentation_id,
+                   roi_id=roi_id)
+
+    def generate_binary_mask_from_threshold(self, threshold: float):
+        """
+        Simple binary mask from a provided threshold.
+        Args:
+            threshold: The threshold to compare values
+
+        Returns:
+
+        Notes:
+            This function will likely become deprecated as we work on more
+            applicable binary thresh holding
+        """
+        return np.where(self._sparse_coo.toarray() >= threshold, 1, 0)
+
+    def generate_roi_inner_edge_coordinates(self, stroke_size: int,
+                                            threshold: float):
+        """
+        Returns a list of coordinates in a video frame for the inner outline
+        of an ROI given a desired stroke size. Looks at each border pixel and
+        moves away from edge by stroke specified pixels. This process is completed
+        by defining the center of the roi and creating a unit vector from the
+        center to the edge point. This unit vector is then multiplied by the stroke
+        and a slight weight to correct for non int centers. The movement vector
+        is added to the edge point and the new point is then set to true in
+        the return mask.
+        Args:
```

Comment:
This is a pretty verbose docstring, and after reading it, I'm still not sure what an `inner outline` is, nor a `stroke size`.
Are you just trying to find the edge of an ROI?
I see you've used contours in a call from here, to respond to my earlier comment.
Can you explain in words what you're trying to do here. I suspect there is some canned routine to just do it.